### PR TITLE
Fix Travis for Python 3.3

### PR DESCRIPTION
**Pull request by isaachier**
Signed-off-by: Isaac Hier <ihier@uber.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -30,7 +30,11 @@ check-virtualenv:
 bootstrap: check-virtualenv install-deps
 
 install-deps:
-	pip install 'setuptools>=20.8.1'
+	if [[ "$(python -V)" == 'Python 3.3.*' ]]; then \
+		setuptools_min_version="18"; \
+	else \
+		setuptools_min_version="20.8.1"; \
```
Comment (yurishkuro) [inline Makefile]: hm, I thought the reason to we pinned to 20.8 was that earlier versions didn't work with some syntax we had.

Comment (isaachier) [reply Makefile]: IDK I just know it doesn't support Python 3.3.

Comment (isaachier) [reply Makefile]: I think this is what introduced the bug in Travis and broke 3.3 builds: https://github.com/pypa/setuptools/commit/114bb58167d4f6462e0b7d69a46233b796840ba8.

Comment (isaachier) [reply Makefile]: Nvm that is a setuptools Travis file, not Travis's copy of setuptools. Here is a similar issue: https://github.com/aws/base64io-python/issues/4.

Comment (isaachier) [reply Makefile]: And this: https://github.com/travis-ci/travis-ci/issues/9002.

Comment (isaachier) [reply Makefile]: OK I just figured it out... I believe that setuptools dropped support for 3.3 in this diff: https://github.com/pypa/setuptools/commit/7392f01ffced3acfdef25b0b2d55cefdc6ee468a. That is when `pip install "setuptools>=20"` stopped working. Release 40 is `>=20`, yet does not support Python 3.3. I might even change the constraint in Python 3.3 to reflect this issue.

File: Makefile [hunk 2]
```diff
@@ -30,7 +30,9 @@ check-virtualenv:
 bootstrap: check-virtualenv install-deps
 
 install-deps:
-	pip install 'setuptools>=20.8.1'
+	# We need to limit to <40.0.0 as long as we support Python 3.3.
+	# See https://github.com/pypa/setuptools/commit/7392f01ffced3acfdef25b0b2d55cefdc6ee468a.
+	pip install 'setuptools>=20.8.1,<40.0.0'
```
Comment (black-adder) [inline Makefile]: can we only add the upperbound restriction if we're using python 3.3?

Comment (isaachier) [reply Makefile]: I tried but makefile syntax is very bad at this.

**Pull request merged**